pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -399,9 +408,27 @@ async def _map_user_prompt(self, part: UserPromptPart) -> list[PartDict]:
                 elif isinstance(item, BinaryContent):
                     # NOTE: The type from Google GenAI is incorrect, it should be `str`, not `bytes`.
                     base64_encoded = base64.b64encode(item.data).decode('utf-8')
-                    content.append({'inline_data': {'data': base64_encoded, 'mime_type': item.media_type}})  # type: ignore
+                    inline_data_dict = {'inline_data': {'data': base64_encoded, 'mime_type': item.media_type}}
+
+                    if item.vendor_metadata:
+                        video_metadata = VideoMetadata(
+                            fps=item.vendor_metadata.get('fps', None),
+                            start_offset=item.vendor_metadata.get('start_offset', None),
+                            end_offset=item.vendor_metadata.get('end_offset', None),
+                        )
+                        inline_data_dict['video_metadata'] = video_metadata  # type: ignore
```

Comment:
Would this work?
```suggestion
                        inline_data_dict['video_metadata'] = item.vendor_metadata  # type: ignore
```

Reply:
I tested it locally and it works, changed as requested